In [80]:
import pandas as pd

In [81]:
flood_all = pd.read_csv('flooding_all.csv')

In [83]:
flood_all.status.value_counts()

Closed          20537
Cancelled         347
Initiated          47
In-progress        27
Name: status, dtype: int64

A: from 2010 to Feburary, 2019, Toronto 311 received 20958 calls complainting a flooding situation. Among those calls, 347 were cancelled soon after by the clients. Therefore, 311 Toronto had dealt with 20611 flooding complaints. Within a duration of 3346 days, the city averagely received 6 flooding complaints a day.

In [84]:
20611/3346

6

## Remove the cancelled calls and create a new data frame named "clean_flood".

In [85]:
clean_flood = flood_all[(flood_all.status == 'Closed') | (flood_all.status == 'Initiated') | (flood_all.status == 'In-progress ')]

Intersections that are more flooded

In [110]:
intersection = clean_flood[clean_flood.first_3_chars_of_postal_code == 'Intersection']

In [115]:
intersection.ADDRESS.value_counts()

Dundas St W & Jane St                                      27
Jane St & St Clair Ave W                                   24
Bloor St E & Parliament St                                 20
Parkside Dr & Bloor St W                                   14
Bloor St W & St George St                                  14
Bloor St E & Castle Frank Rd                               12
The Queensway & Parkside Dr                                12
Don Mills Rd & York Mills Rd                               11
Wynford Dr Eglinton W Ramp & Eglinton E Wynford Dr Ramp    11
Royal York S Dundas E Ramp & Royal York N Dundas W Ramp    10
Madelaine Ave & Danforth Ave                               10
Seaton St & Shuter St                                       9
Bathurst St & Lawrence Ave W                                9
William Carson Cres & Yonge St                              9
Eglinton Ave E & Mount Pleasant Rd                          8
Clendenan Ave & Bloor St W                                  8
Dundas S

Q: Which year received the most flooding calls?

In [87]:
clean_flood.year.value_counts()

2014    5042
2013    2608
2015    2457
2018    2060
2017    1809
2011    1714
2012    1389
2016    1240
2019    1235
2010    1057
Name: year, dtype: int64

Q: Which month is the busiest?

In [88]:
clean_flood.month.value_counts()

2     3951
6     2217
10    1716
1     1710
3     1708
8     1699
7     1644
5     1601
4     1468
9     1155
11     992
12     750
Name: month, dtype: int64

Q: How the breakdowns by zipcode? 

In [90]:
code_breadown = clean_flood.CODE_clean.value_counts()


In [103]:
code_breadown.head(10)

M6S    531
M6N    515
M1C    396
M1N    381
M6H    379
M3H    373
M5M    372
M1E    372
M4J    363
M4L    361
Name: CODE_clean, dtype: int64

In [91]:
code_breadown.to_csv('input/code_breakdown.csv')

/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ipykernel_launcher.py:1: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  """Entry point for launching an IPython kernel.


Q: How the breakdowns by wards?

In [92]:
ward_breakdown = clean_flood.ward.value_counts()

In [105]:
ward_breakdown.head(20)

Eglinton-Lawrence (16)          746
Etobicoke-Lakeshore (05)        684
Parkdale-High Park (13)         634
York South-Weston (11)          613
Don Valley West (25)            571
Trinity-Spadina (20)            568
Beaches-East York (32)          544
Toronto Centre-Rosedale (27)    539
Toronto-Danforth (30)           537
Scarborough Southwest (36)      532
Scarborough East (44)           531
Davenport (17)                  467
Scarborough Centre (37)         463
Etobicoke Centre (04)           463
Trinity-Spadina (19)            443
Eglinton-Lawrence (15)          443
St. Paul's (21)                 432
Toronto Centre-Rosedale (28)    431
Etobicoke-Lakeshore (06)        414
St. Paul's (22)                 409
Name: ward, dtype: int64

In [93]:
ward_breakdown.to_csv('input/ward_breakdown.csv')

/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ipykernel_launcher.py:1: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  """Entry point for launching an IPython kernel.


Q: Which day received the most calls and what happened that day?

In [94]:
clean_flood.date.value_counts()

2014-02-21    463
2014-02-20    448
2014-02-19    332
2014-10-16    323
2014-01-11    272
2019-02-15    230
2015-10-28    228
2013-02-11    144
2014-01-10    126
2014-03-11    125
2019-02-04    121
2013-02-28    121
2019-01-23    115
2019-02-24    112
2014-02-22    105
2019-02-03    105
2018-04-16    100
2015-12-29     99
2013-07-08     97
2015-06-08     90
2017-05-25     88
2014-03-10     86
2017-05-05     86
2015-03-04     85
2019-02-21     83
2013-02-19     82
2014-02-18     81
2013-02-27     80
2012-09-04     79
2017-05-01     79
             ... 
2016-03-12      1
2012-11-16      1
2017-03-14      1
2017-10-02      1
2010-07-18      1
2016-08-22      1
2011-02-25      1
2016-11-27      1
2010-02-24      1
2010-05-18      1
2016-02-29      1
2013-02-23      1
2010-10-29      1
2016-05-09      1
2016-05-08      1
2011-07-24      1
2013-09-08      1
2011-07-21      1
2010-03-11      1
2011-10-10      1
2015-09-13      1
2013-09-01      1
2017-02-23      1
2012-04-24      1
2016-08-20

Q: How the calls distribute in the worst zipcode?

In [95]:
worst_zip = clean_flood[clean_flood.CODE_clean == 'M6S']

In [96]:
worst_zip.date.value_counts()

2014-02-19    14
2014-02-20    11
2019-02-15     6
2014-03-11     6
2014-03-07     6
2016-11-03     6
2015-06-27     5
2015-10-28     5
2019-02-04     5
2018-02-20     5
2017-01-12     5
2014-02-21     5
2018-01-12     4
2012-07-31     4
2014-03-10     4
2014-03-20     4
2014-01-11     4
2013-06-13     4
2019-01-23     4
2017-01-17     4
2014-10-16     4
2015-12-29     4
2014-09-21     3
2013-02-27     3
2018-05-15     3
2012-05-09     3
2010-06-02     3
2012-07-16     3
2012-07-17     3
2016-02-24     3
              ..
2015-07-09     1
2018-08-05     1
2018-06-18     1
2019-02-08     1
2018-12-03     1
2015-02-23     1
2010-01-25     1
2010-11-16     1
2017-04-04     1
2011-06-21     1
2015-02-11     1
2014-12-12     1
2018-01-22     1
2016-03-11     1
2016-03-15     1
2015-04-20     1
2018-08-06     1
2019-01-02     1
2011-08-11     1
2016-10-16     1
2014-02-24     1
2014-02-25     1
2019-02-06     1
2010-11-23     1
2016-11-16     1
2019-01-30     1
2010-11-24     1
2015-02-28    

Q: How the calls distributed in the worst ward?

In [97]:
worst_ward = clean_flood[clean_flood.ward == 'Eglinton-Lawrence (16)']

In [98]:
worst_ward.date.value_counts()

2014-02-21    32
2014-02-19    23
2014-10-16    21
2014-01-11    18
2014-02-20    18
2015-10-28    16
2014-01-10    12
2013-11-17     7
2012-09-04     6
2014-02-22     6
2011-08-22     6
2014-03-14     6
2015-03-09     5
2013-07-08     5
2015-02-25     5
2014-04-29     5
2014-01-13     5
2011-11-29     5
2011-04-27     5
2011-02-14     5
2015-06-10     5
2013-05-29     5
2015-03-04     4
2015-12-29     4
2013-02-27     4
2011-03-10     4
2011-08-21     4
2014-10-17     4
2013-06-11     4
2013-02-19     4
              ..
2017-04-07     1
2011-02-23     1
2013-06-13     1
2017-04-03     1
2013-06-10     1
2015-02-07     1
2013-06-19     1
2013-02-12     1
2013-02-10     1
2010-08-31     1
2011-02-27     1
2010-06-23     1
2011-12-16     1
2015-07-17     1
2010-06-28     1
2017-06-08     1
2017-10-19     1
2012-07-31     1
2013-01-13     1
2015-12-22     1
2014-06-25     1
2014-09-29     1
2014-06-26     1
2011-02-06     1
2013-04-24     1
2016-01-12     1
2016-01-15     1
2014-09-21    

How many days in 10 years, the city had seen more than 100 complaints?

In [99]:
days = pd.read_csv('output/merged_new.csv')
days.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3346 entries, 0 to 3345
Data columns (total 5 columns):
date            3346 non-null object
year            3346 non-null int64
month/day       3346 non-null object
counts          3346 non-null int64
Total_Precip    3223 non-null float64
dtypes: float64(1), int64(2), object(2)
memory usage: 130.8+ KB


In [102]:
days[days.counts>100].sort_values('counts', ascending=False)

date  year month/day  counts  Total_Precip
1512  2014-02-21  2014    21-Feb     467           8.7
1511  2014-02-20  2014    20-Feb     452          13.4
1510  2014-02-19  2014    19-Feb     334           0.2
1749  2014-10-16  2014    16-Oct     333          21.1
1471  2014-01-11  2014    11-Jan     278           9.6
2126  2015-10-28  2015    28-Oct     232          58.3
3332  2019-02-15  2019    15-Feb     231           1.3
1137  2013-02-11  2013    11-Feb     147           5.6
1470  2014-01-10  2014    10-Jan     129           2.0
1530  2014-03-11  2014    11-Mar     126           0.0
3321  2019-02-04  2019    04-Feb     124           2.1
1154  2013-02-28  2013    28-Feb     122           3.9
3309  2019-01-23  2019    23-Jan     118          11.0
3341  2019-02-24  2019    24-Feb     112          10.0
1513  2014-02-22  2014    22-Feb     106           0.0
3320  2019-02-03  2019    03-Feb     105           0.0
3027  2018-04-16  2018    16-Apr     101          28.8